src/utils/Buffer.cpp

```diff
@@ -90,3 +91,12 @@ std::string Buffer::consume(long bytes)
   remove(bytes);
   return newStr;
 }
+
+std::string Buffer::toString()
+{
+  std::stringstream oss;
+  for (std::size_t i = 0; i < _buff.size(); ++i) {
+    oss << static_cast<char>(_buff[i]);
+  }
+  return oss.str();
+}
```

Comment:
```suggestion
std::string Buffer::toString()
{
  return std::string(_buff.begin(), _buff.end());
}
```
Implicit conversion from `unsigned char` to `char` is fine. Cutting out the `std::stringstream` will also be faster.